watertap/unit_models/boron_removal.py

```diff
@@ -766,13 +772,23 @@ def initialize_build(
         # Solve unit
         with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
             res = opt.solve(blk, tee=slc.tee)
+            # occasionally it might be worth retrying a solve
+            if not check_optimal_termination(res):
+                init_log.warning(
+                    f"Trouble solving unit model {blk.name}, trying one more time"
+                )
+                res = opt.solve(blk, tee=slc.tee)
         init_log.info_high("Initialization Step 2 {}.".format(idaeslog.condition(res)))
 
         # ---------------------------------------------------------------------
         # Release Inlet state
         blk.control_volume.release_state(flags, outlvl + 1)
         init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
 
+        if not check_optimal_termination(res):
+            if raise_on_failure:
```

Comment:
I am not sure if this is a good idea either - the whole purpose of the InitializationError is to force the user to recognize it and handle it. This makes it a bit too easy to suppress and ignore these,